boards/arm/nrf5340_audio_dk_nrf5340/board.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef __BOARD_H__
+#define __BOARD_H__
+
+#include <zephyr.h>
+
+/* Voltage divider PCA10121 board versions.
+ * The defines give what value the ADC will read back.
+ * This is determined by the on-board voltage divider.
+ */
+
+struct board_version {
+	char name[10];
+	uint32_t mask;
+	uint32_t adc_reg_val;
+};
+
+#define BOARD_PCA10121_0_0_0_MSK (BIT(0))
+#define BOARD_PCA10121_0_6_0_MSK (BIT(1))
+#define BOARD_PCA10121_0_7_0_MSK (BIT(2))
+#define BOARD_PCA10121_0_7_1_MSK (BIT(3))
+#define BOARD_PCA10121_0_8_0_MSK (BIT(4))
+#define BOARD_PCA10121_0_8_1_MSK (BIT(5))
+#define BOARD_PCA10121_0_8_2_MSK (BIT(6))
+#define BOARD_PCA10121_0_9_0_MSK (BIT(7))
+#define BOARD_PCA10121_0_10_0_MSK (BIT(8))
+#define BOARD_PCA10121_1_0_0_MSK (BIT(9))
+
+static const struct board_version BOARD_VERSION_ARR[] = {
+	{ "0.0.0", BOARD_PCA10121_0_0_0_MSK, INT_MIN },
+	{ "0.6.0", BOARD_PCA10121_0_6_0_MSK, 61 },
+	{ "0.7.0", BOARD_PCA10121_0_7_0_MSK, 102 },
+	{ "0.7.1", BOARD_PCA10121_0_7_1_MSK, 303 },
+	{ "0.8.0", BOARD_PCA10121_0_8_0_MSK, 534 },
+	{ "0.8.1", BOARD_PCA10121_0_8_1_MSK, 780 },
+	{ "0.8.2", BOARD_PCA10121_0_8_2_MSK, 1018 },
+	{ "0.9.0", BOARD_PCA10121_0_9_0_MSK, 1260 },
+	/* Lower value used on 0.10.0 due to high ohm divider */
+	{ "0.10.0", BOARD_PCA10121_0_10_0_MSK, 1480 },
+	{ "1.0.0", BOARD_PCA10121_1_0_0_MSK, 1743 },
+};
+
+#define BOARD_VERSION_VALID_MSK                                                                    \
+	(BOARD_PCA10121_0_7_0_MSK | BOARD_PCA10121_0_7_1_MSK | BOARD_PCA10121_0_8_0_MSK |          \
+	 BOARD_PCA10121_0_8_1_MSK | BOARD_PCA10121_0_8_2_MSK | BOARD_PCA10121_0_9_0_MSK |          \
+	 BOARD_PCA10121_0_10_0_MSK | BOARD_PCA10121_1_0_0_MSK)
+
+#define BOARD_VERSION_VALID_MSK_SD_CARD                                                            \
+	(BOARD_PCA10121_0_7_0_MSK | BOARD_PCA10121_0_8_0_MSK | BOARD_PCA10121_0_8_1_MSK |          \
+	 BOARD_PCA10121_0_8_2_MSK | BOARD_PCA10121_0_9_0_MSK | BOARD_PCA10121_0_10_0_MSK |         \
+	 BOARD_PCA10121_1_0_0_MSK)
+
+#define BOARD_REVISION_VALID_MSK_MAX14690_PMIC                                                     \
+	(BOARD_PCA10121_0_6_0_MSK | BOARD_PCA10121_0_7_0_MSK | BOARD_PCA10121_0_7_1_MSK)
+
+#define LED_APP_RGB 0
+#define LED_NET_RGB 1
+#define LED_APP_1_BLUE 2
+#define LED_APP_2_GREEN 3
+#define LED_APP_3_GREEN 4
```

Comment:
can this be explained? why is this required?

Reply:
The Audio DK exists in a number of revisions, many of which are still supported by us. In order for the FW to know which version it is running on, and make appropriate adjustments these defines are required.

Reply:
I take the liberty of resolving this issue with no action. Please voice if you want further discussion or changes.

Reply:
why don't we have board revisions (supported by Zephyr) and store this information in Devicetree per-revision? Also, what are all the `LED_*` defines?

Reply:
I agree with @gmarull; that is exactly what per-revision devicetree overlays are for. See https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/porting/board_porting.html#multiple-board-revisions for details. This was actually something that Nordic added to Zephyr, so let's please use it.

Reply:
@gmarull @mbolivar-nordic As far as I can see the board-revision overlays are for using different Kconfig settings and dts-changes based on the version you are compiling for. What we are doing, and the reason for needing these values, is checking the board version through a voltage divider. This means we compile one hex and it can work on multiple different boards. What it seems like the board-revision overlays does is making sure you compile for one specific board version only. 
Also, the ADC values still would need to be stored here, right?

Reply:
I agree (perhaps not surprisingly :P ) with Alexander. There could be a situation such as a demo or showroom floor where someone needs to flash N kits, where not all kits are of the same HW version. This way, the user can flash the same hex to all boards and the FW will adapt automatically. If such a feature exists, then please point it out. Anyway, we ask that we can make such a change after the merge.

Reply:
I see, OK, that sounds fine then. Sorry for the misunderstanding.